spec/spec_helper.rb

```diff
@@ -24,6 +24,35 @@ def pattern_path(path)
 require "logstash/filters/grok"
 
 module GrokHelpers
+  module Shared
```

Comment:
naming: it is not clear that `Shared` is all about a shared _pattern mode_. Maybe name it `SharedPatternMode` to make it more clear and less likely to become a catch-all for future shared state?

Reply:
renamed to `PatternModeSupport`